query/src/main/scala/filodb/query/exec/InProcessPlanDispatcher.scala

```diff
@@ -35,12 +37,12 @@ import filodb.query.Query.qLogger
     Kamon.runWithSpan(Kamon.currentSpan(), false) {
       // translate implicit ExecutionContext to monix.Scheduler
       val querySession = QuerySession(plan.execPlan.queryContext, queryConfig, catchMultipleLockSetErrors = true)
-      plan.execPlan.execute(source, querySession).onErrorRecover {
-        case e: AskTimeoutException if (plan.execPlan.queryContext.plannerParams.allowPartialResults)
+      plan.execPlan.execute(source, querySession).timeout(plan.clientParams.deadline.milliseconds).onErrorRecover {
+        case e: TimeoutException if (plan.execPlan.queryContext.plannerParams.allowPartialResults)
         =>
           qLogger.warn(s"Swallowed AskTimeoutException for query id: ${plan.execPlan.queryContext.queryId} " +
             s"since partial result was enabled: ${e.getMessage}")
```

Comment:
I played around with this, and I don't think we need to explicitly handle `AskTimeout`s in `InProcessDispatcher`.

If I:
- locally change `ActorPlanDispatcher`'s `ask` `Future` to a `Future` that immediately throws an `AskTimeoutException`
- remove all of this timeout-handler code

then the new test passes.

[Here](https://github.com/filodb/FiloDB/files/9195675/patch.txt) is the patch for these changes. (Most importantly, the test `execPlan[1,2]` `QueryContext`s are missing `allowPartialResults = true`).